scripts/maintenance-status/cloud-create-mr

```diff
@@ -0,0 +1,146 @@
+    return run_cmd(f"{IOSC} rdiff {old_prj} {pkg} {new_prj}")
+
+
+def iosc_results(project, pkg):
+    return run_cmd(f"{IOSC} results -r "
+                   f"SLE_12_SP{CLOUD_SLESSP_VERSION[CLOUD_VERSION]} "
+                   f"-a x86_64 {project} {pkg}")
+
+
+def get_build_status(project, package):
+    build_status = iosc_results(project, package).split()[-1]
+    while(build_status not in ['succeeded*', 'failed', 'unresolvable',
+                               'unresolvable*', 'locked', 'disabled']):
+        print(build_status)
+        if(build_status in ['scheduled', 'building*', 'building', 'blocked']):
+            build_status = iosc_results(project, package).split()[-1]
+            print(f"  Waiting for package to build: {build_status}")
+            time.sleep(30)
+    return build_status
+
+
+exclude_regex = "(" + ")|(".join(EXCLUDES) + ")"
+
+product_pkgs = {i for i in iosc_ls(
+    PRODUCT).split() if not re.match(exclude_regex, i)}
+update_pkgs = {i for i in iosc_ls(
+    UPDATE).split() if not re.match(exclude_regex, i)}
+
+
+updated_pkgs = {}
+new_pkgs = {}
+build_failed_pkgs = []
+for source in SOURCES:
+    source_pkgs = {i for i in iosc_ls(
+        source).split() if not re.match(exclude_regex, i)}
+    common_product_pkgs = source_pkgs & product_pkgs
+    common_update_pkgs = source_pkgs & update_pkgs
+    new_pkgs[source] = list((source_pkgs - product_pkgs) - update_pkgs)
+
+    for pkg in sorted(common_product_pkgs | common_update_pkgs):
+        if pkg in common_update_pkgs:
+            old_prj = UPDATE
+        else:
+            old_prj = PRODUCT
+        print(f"processing package: {pkg}")
+        diff = iosc_rdiff(old_prj, pkg, source)
+        diff_changes = re.search(
+            fr"^Index:\s{pkg}.changes.+^[\+\s]------+", diff,
+            re.MULTILINE | re.DOTALL)
+        if (not(diff and diff_changes)):
+            continue
+        build_status = get_build_status(source, pkg)
+        if ('succeeded' in build_status):
+            if source not in updated_pkgs:
+                updated_pkgs[source] = {}
+            updated_pkgs[source][pkg] = diff_changes.group()
+        else:
+            build_failed_pkgs.append(pkg)
+        print(f"  - updated (build: {build_status})")
+
+
+print(f"""
+####################################################################################################
+                      Summary for: {SOURCES}
+####################################################################################################
+
+    Updated:
+{indent(yaml.dump(sorted(list({k: v for l in updated_pkgs.values() for k,v in l.items()})), default_flow_style=False), 6 * ' ')}
+    Updated but failing to build:
+{indent(yaml.dump(build_failed_pkgs, default_flow_style=False), 6 * ' ')}
+    New:
+{indent(yaml.dump(new_pkgs, default_flow_style=False), 6 * ' ')}
+    Excluded:
+{indent(yaml.dump(EXCLUDES, default_flow_style=False), 6 * ' ')}
+""")
+
+# TODO: FIX showing diff for packges
```

Comment:
I think that there is also another TODO: you need to check that the package is in one of the _product group.

This can explain one bug that @gosipyan found recently, where openstack-quickstart was submitted as part of a MR.

To do that I suggest to read all the group files, for example via `iosc cat Devel:Cloud:8/_product/FILE.group`, merge all together, and check with `if f'"{pkg}" supportstatus="l3"' in produc:`

Reply:
Thanks @aplanas, this is something I was not aware.

I have added that check as you suggested!